test/providers/common_tests.dart

```diff
@@ -282,12 +282,12 @@ commonProviderTests(RouterFactory routerFactory) {
   });
 
   group('hierarchical routing', () {
-    void _testParentChild(Pattern parentPath, Pattern childPath,
+    Future<bool> _testParentChild(Pattern parentPath, Pattern childPath,
```

Comment:
only needs to be marked async if you're using await within the method
don't have to if you're directly returning the future